### ISSUE TITLE
Segment control very slow and unresponsive in IOS

### ISSUE THREAD

**Issue by ajaybeniwal** [driftyco/ionic2#663]
I am creating one view with segment control but it is very slow on IOS9 to switch between the segments ,it is really slow first time when the view loads after that it is not very slow but  little slow in general.

**Comment by adamdbradley** [comment]
We'll need more details. What does your markup look like? What does your JS look like? What's being loaded in each segment control?

**Comment by ajaybeniwal** [comment]
```<ion-navbar *navbar>
  <a menu-toggle>
    <icon menu></icon>
  </a>
  <ion-title>Inbound Shipment</ion-title>
</ion-navbar>

<ion-content>
  <ion-segment [(ng-model)]="shipmentShowing">
    <ion-segment-button value="carrier">
      Carrier
    </ion-segment-button>
    <ion-segment-button value="handcarry">
      Hand Carry
    </ion-segment-button>
    <ion-segment-button value="pending">
      Pending
    </ion-segment-button>
  </ion-segment>

  <div [ng-switch]="shipmentShowing">


    <ion-list *ng-switch-when="'carrier'">
      <ion-item-group>
        <ion-item-divider>
          Location Details
        </ion-item-divider>
        <button ion-item (click)="getIBX()">
          IBX
          <ion-note item-right>
            Required
          </ion-note>
        </button>
        <button ion-item (click)="getCage()">
          Cage/Suite
          <ion-note item-right>
            Required
          </ion-note>
        </button>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <button ion-item (click)="getIBX()">
          Scheduling
          <ion-note item-right>ou Required
          </ion-note>
        </button>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
          SHIPMENT DETAILS
        </ion-item-divider>
        <button ion-item (click)="getIBX()">
          Shipping Carrier
          <ion-note item-right>
            Required
          </ion-note>
        </button>
        <button ion-item (click)="getIBX()">
          Number of Boxes
          <ion-note item-right>
            Required
          </ion-note>
        </button>
        <ion-switch item-right [(ng-model)]="oversizedShipment" [(checked)]="oversizedShipment" secondary>
          Oversized Shipment
        </ion-switch>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
          Additional IBX Service
        </ion-item-divider>
        <ion-switch item-right [(ng-model)]="deliverPackage" [(checked)]="deliverPackage" secondary>
          Deliver Package(s)
        </ion-switch>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <ion-input>
          <input type="text" placeholder="Additional Request">
        </ion-input>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <ion-input>
          <input type="text" placeholder="Purchase Order Reference">
        </ion-input>
        <ion-input>
          <input type="text" placeholder="Customer Internal Reference">
        </ion-input>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <button ion-item (click)="redirectTechnicalContact()">
          Technical Contact
        </button>
      </ion-item-group>
    </ion-list>
    <ion-list *ng-switch-when="'handcarry'">
      <ion-item-group>
        <ion-item-divider>
          Location Details
        </ion-item-divider>
        <button ion-item (click)="getIBX()">
          IBX
          <ion-note item-right>
            Required
          </ion-note>
        </button>
        <button ion-item (click)="getCage()">
          Cage/Suite
          <ion-note item-right>
            Required
          </ion-note>
        </button>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <button ion-item (click)="getIBX()">
          Scheduling
          <ion-note item-right>ou Required
          </ion-note>
        </button>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
          SHIPMENT DETAILS
        </ion-item-divider>
        <button ion-item (click)="getIBX()">
          Shipping Carrier
          <ion-note item-right>
            Required
          </ion-note>
        </button>
        <button ion-item (click)="getIBX()">
          Number of Boxes
          <ion-note item-right>
            Required
          </ion-note>
        </button>
        <ion-switch item-right [(ng-model)]="oversizedShipment" [(checked)]="oversizedShipment" secondary>
          Oversized Shipment
        </ion-switch>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
          Additional IBX Service
        </ion-item-divider>
        <ion-switch item-right [(ng-model)]="deliverPackage" [(checked)]="deliverPackage" secondary>
          Deliver Package(s)
        </ion-switch>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <ion-input>
          <input type="text" placeholder="Additional Request">
        </ion-input>
      </ion-item-group>
      <ion-item-group>
        <ion-item-divider>
        </ion-item-divider>
        <ion-input>
          <input type="text" placeholder="Purchase Order Reference">
        </ion-input>
        <ion-input>
          <input type="text" placeholder="Customer Internal Reference">
        </ion-input>
      </ion-item-group>
    </ion-list>
    <ion-list *ng-switch-when="'pending'">
    <ion-item-group>
    <ion-item-divider class="no-results">
      No pending storage items
    </ion-item-divider>
  </ion-item-group>
    </ion-list>
    <div padding>
      <button block (click)="signin()">Submit Request</button>
    </div>
  </div>

</ion-content>```

**Comment by ajaybeniwal** [comment]
///<referenc path="../../../typings/angular2/angular2.d.ts" />
import {Page, NavController, ViewController} from 'ionic/ionic';
import {TechnicalContact} from '../technicalcontact/technicalcontact';
@Page({
  templateUrl: 'app/inboundshipment/inboundshipment.html',
})
export class InboundShipment {
  constructor(nav: NavController, view: ViewController) {
    this.nav = nav;
    this.view = view;
     this.shipmentShowing = 'carrier';
  }
  
  redirectTechnicalContact(){
    this.nav.push(TechnicalContact, {
    });
  }

}

**Comment by astec** [comment]
+1

**Comment by mlynch** [comment]
Thanks for the report, this has been fixed.

**Comment by astec** [comment]
Thanks @mlynch, any chance to get a fix for #657 (*Segment related*) in nearest time?
